### PR TITLE
squeak: 4.10.2.2614 -> 202003021730

### PR DESCRIPTION
**Pull request by BenBals**
###### Motivation for this change
The Squeak package has not been updated for quite some time. I have updated it to use the opensmalltalk repo where the current Squeak development happens.

This is my first substantial contribution. I am happy to receive feedback and improve it. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/squeak/default.nix [hunk 1]
```diff
@@ -1,33 +1,55 @@
...
+  version = "202003021730";
 
-  src = fetchurl {
-    sha256 = "0bpwbnpy2sb4gylchfx50sha70z36bwgdxraym4vrr93l8pd3dix";
-    url = "http://squeakvm.org/unix/release/Squeak-${version}-src.tar.gz";
+  src = fetchFromGitHub {
+    owner="OpenSmalltalk";
+    repo="opensmalltalk-vm";
+    rev="${version}";
+    sha256 = "0qxnmw5q836m09y7pkii1dwkjsjwa86pyhwxj11hxsf815r15c0v";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/squeak/default.nix:13]:
```suggestion
    sha256 = "1cs0vmza1nq03i5f6w3j6l78j76qwlmc2aakvc10ivmy4vknglmc";
```

File: pkgs/development/compilers/squeak/default.nix [hunk 2]
```diff
@@ -1,33 +1,55 @@
-{ stdenv, fetchurl, cmake, coreutils, dbus, freetype, glib, gnused
-, libpthreadstubs, pango, pkgconfig, libpulseaudio, which }:
+{ stdenv, fetchFromGitHub, lib, alsaLib, autoconf, automake, bash, clang, coreutils
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/squeak/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, alsaLib, autoconf, automake, bash, clang, coreutils
```

File: pkgs/development/compilers/squeak/default.nix [hunk 3]
```diff
@@ -1,33 +1,55 @@
...
-  version = "4.10.2.2614";
+  version = "202003021730";
 
-  src = fetchurl {
-    sha256 = "0bpwbnpy2sb4gylchfx50sha70z36bwgdxraym4vrr93l8pd3dix";
-    url = "http://squeakvm.org/unix/release/Squeak-${version}-src.tar.gz";
+  src = fetchFromGitHub {
+    owner="OpenSmalltalk";
+    repo="opensmalltalk-vm";
+    rev="${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/squeak/default.nix:12]:
```suggestion
    owner = "OpenSmalltalk";
    repo = "opensmalltalk-vm";
    rev = version;
```

File: pkgs/development/compilers/squeak/default.nix [hunk 4]
```diff
@@ -1,33 +1,55 @@
...
-  '';
+  ldLibraryPath = stdenv.lib.makeLibraryPath  [
+    libuuid
+    xorg.libX11
+  ];
 
-  configurePhase = ''
-    unix/cmake/configure --prefix=$out --enable-mpg-{mmx,pthreads}
+  buildPhase = ''
+   find . -type f -exec sed -i -e 's/\/usr\/bin\/env/${lib.escape ["/"] coreutils.outPath}\/bin\/env/g' {} \;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/squeak/default.nix:40]:
```suggestion
   find . -type f -exec sed -i -e 's/\/usr\/bin\/env/${stdenv.lib.escape ["/"] coreutils.outPath}\/bin\/env/g' {} \;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Please squash the commits together.

**Comment by ehmry** [comment]
Closed because this is continued in #133310

**Pull request closed without merging**